method--pull-and-process-acs5.R

```diff
@@ -0,0 +1,827 @@
+    table_meta <- 
+      table_meta %>% 
+      mutate(pov_status =
+               case_when(str_detect(label, "at or above poverty") ~ "NonPovInc",
+                         str_detect(label, "below poverty")       ~ "PovInc") %>% 
+               replace_na("All"))
+  }
+  
+  # Recode Presence of own children --------------------------------------------
+  if (any(str_detect(my_concepts, "PRESENCE OF OWN CHILDREN UNDER 18"))) {
+    #table_meta <- filter(my_meta, str_detect(name, "B23007"))
+    table_meta <- 
+      table_meta %>% 
+      mutate(own_kids_under18 =
+               case_when(str_detect(label, "With own children under 18 years") ~ "SomeKidsUnder18",
+                         str_detect(label, "No children under 18 years")       ~ "NoKidsUnder18") %>% 
+               replace_na("All"))
+  }
+  
+  # Recode Grandparent Presence ------------------------------------------------
+  if (any(str_detect(my_concepts, "GRANDPARENTS LIVING WITH OWN GRANDCHILDREN"))) {
+    #table_meta <- filter(my_meta, str_detect(name, "B10051"))
+    table_meta <- 
+      table_meta %>% 
+      mutate(gp_resp = 
+               case_when(str_detect(label, "Grandparent not responsible") ~ "GParNotResponsible",
+                         str_detect(label, "Grandparent responsible")     ~ "GParIsResponsible" ) %>% 
+               replace_na("All"))
+    if (any(str_detect(my_concepts, "BY PRESENCE OF PARENT"))) {
+      table_meta <- 
+        table_meta %>% 
+        mutate(gp_parent = 
+                 case_when(str_detect(label, "no parent of grandchildren") ~ "NoParentPresent",
+                           str_detect(label, "Other grandparents")         ~ "OtherGPPresent" ) %>% 
+                 replace_na("All"))
+    }
+  }
+  
+  # Recode Labor Force, Employment Status, and Family Type ---------------------
+  if (any(str_detect(my_concepts, "FAMILY TYPE|EMPLOYMENT"))) { 
+    if (any(str_detect(my_concepts, "FAMILY TYPE BY EMPLOYMENT"))) {
+      table_meta <- 
+        table_meta %>% 
+        mutate(
+          family_type = 
+            case_when(str_detect(label, "[Mm]arried-couple family")         ~ "Married",
+                      str_detect(label, "[Ff]emale householder, no spouse") ~ "UnmarriedFemaleHh",
+                      str_detect(label, "[Mm]ale householder, no spouse")   ~ "UnmarriedMaleHh",
+                      str_detect(label, "[Oo]ther family")                  ~ "Unmarried") %>% 
+            replace_na("All"),
+          lfp = 
+            case_when(
+              family_type == "Married" ~
+                case_when(str_detect(label, "Husband in.+Wife in")         ~ "HusbandInLfpWifeInLfp",
+                          str_detect(label, "Husband in.+Wife not in")     ~ "HusbandInLfpWifeNotInLfp",
+                          str_detect(label, "Husband not in.+Wife in")     ~ "HusbandNotInLfpWifeInLfp",
+                          str_detect(label, "Husband not in.+Wife not in") ~ "HusbandNotInLfpWifeNotInLfp",
```

Comment:
Add the following:
str_detect(label, "Wife in")                   ~ "WifeInLfp",
str_detect(label, "Wife not in")             ~ "WifeNotInLfp",

Reply:
I have other code running, making me unable to easily check this at the moment, but I don't believe that your proposed recodings would impact how things recode in practice. The way that the [metadata for this table](https://www.socialexplorer.com/data/ACS2015_5yr/metadata/?ds=ACS15_5yr&table=B23007) is organized never has a status describe only for wife, in married couples. It's always the husband's status that is described first. 🙄

Fwiw, I believe that the next best modification to this section of code is moving to [Table B23008](https://www.socialexplorer.com/data/ACS2015_5yr/metadata/?ds=ACS15_5yr&table=B23008) which is much simpler to code and review.